pkg/cstor/cstor.go

```diff
@@ -427,57 +431,32 @@ func (p *Plugin) CreateVolumeFromSnapshot(snapshotID, volumeType, volumeAZ strin
 	volumeID := s[0]
 	snapName := s[1]
 
-	p.Log.Infof("Restoring snapshot{%s} for volume:%s", snapName, volumeID)
-
-	newVol, e := p.createPVC(volumeID, snapName)
-	if e != nil {
-		return "", errors.Errorf("Failed to restore PVC.. %s", e)
-	}
-
-	p.Log.Infof("New volume(%v) created", newVol)
-
-	restoreSpec := &v1alpha1.CStorRestoreSpec{
-		RestoreName: newVol.backupName,
-		VolumeName:  newVol.volname,
-		RestoreSrc:  p.cstorServerAddr,
-	}
-
-	restore := &v1alpha1.CStorRestore{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: newVol.namespace,
-		},
-		Spec: *restoreSpec,
+	snapType := "cloud"
+	if p.local {
+		snapType = "local"
 	}
 
-	url := p.mayaAddr + restorePath
+	p.Log.Infof("Restoring %s snapshot{%s} for volume:%s", snapType, snapName, volumeID)
 
-	restoreData, err := json.Marshal(restore)
+	newVol, err := p.getVolInfo(volumeID, snapName)
 	if err != nil {
-		p.Log.Errorf("Error during JSON marshal : %s", err.Error())
-		return "", err
+		return "", errors.Wrapf(err, "Failed to read PVC")
```

Comment:
better to print names also.. not only here, in createPVC also..
but, in order NOT to dilute this PR, it can taken for next PR..

Reply:
ok.